pkg/lifecycle/lifecycle.go

```diff
@@ -114,25 +118,28 @@ type Transition struct {
 	XMLName      xml.Name       `xml:"Transition" json:"-"`
 	Date         ExpirationDate `xml:"Date,omitempty" json:"Date,omitempty"`
 	StorageClass string         `xml:"StorageClass,omitempty" json:"StorageClass,omitempty"`
-	Days         ExpirationDays `xml:"Days,omitempty" json:"Days,omitempty"`
+	Days         ExpirationDays `xml:"Days" json:"Days"`
 }
 
 // MarshalJSON customizes json encoding by omitting empty values
 func (t Transition) MarshalJSON() ([]byte, error) {
+	if t.IsNull() {
+		return nil, nil
+	}
 	type transition struct {
 		Date         *ExpirationDate `json:"Date,omitempty"`
 		StorageClass string          `json:"StorageClass,omitempty"`
-		Days         *ExpirationDays `json:"Days,omitempty"`
+		Days         *ExpirationDays `json:"Days"`
```

Comment:
This cannot be `omitempty` - either transition-days or transition-date will be specified